solution 04-04-layouts

aleksandrayakovleva132 opened the pull request.

aleksandrayakovleva132 committed:
solution 04-04-layouts

jsru commented:
Добавляю преподавателя (@ShGKme) для код-ревью.

aleksandrayakovleva132 committed:
layouts solution

jsru commented:
Решение было обновлено, посмотрим что скажет @ShGKme

aleksandrayakovleva132 committed:
Merge branch 'master' of github.com:js-tasks-ru/vue-20210304_aleksandra-yakovleva9

ShGKme commented on 04-spa/04-Layouts/components/FormLayout.vue:
```diff
@@ -1,8 +1,19 @@
-<template></template>
+<template>
+  <section class="page page_meetup-forms">
+    <h2>{{ title }}</h2>
+    <slot />
+  </section>
```
Вёрстка чуть-чуть не совпадает: https://github.com/js-tasks-ru/vue-course-materials/blob/master/meetups-markup/meetup-form.html#L29-L31

ShGKme commented on 04-spa/04-Layouts/components/AuthLayout.vue:
```diff
@@ -1,8 +1,18 @@
-<template></template>
-
+<template>
+  <section class="page page_onboarding">
+    <h2>{{ title }}</h2>
+    <slot />
+  </section>
```
Вёрстка чуть-чуть не совпадает:
https://github.com/js-tasks-ru/vue-course-materials/blob/master/meetups-markup/login.html#L30-L32

ShGKme commented on 04-spa/04-Layouts/components/TheHeader.vue:
```diff
@@ -1,4 +1,19 @@
+  <header class="header">
+    <div>
+      <h1>
+        <a href="/" class="router-link-exact-active router-link-active"
+        ><img src="img/logo.svg" alt="Meetups"
+        /></a>
```
Здесь также лучше сделать `router-link`

ShGKme commented on 04-spa/04-Layouts/components/TheHeader.vue:
```diff
@@ -1,4 +1,19 @@
+      </h1>
+    </div>
+    <nav>
+      <router-link :to="{ name: 'meetups' }">Митапы</router-link> |
+      <router-link :to="{ name: 'form' }">Создать митап</router-link> |
+      <router-link :to="{ name: 'login' }">Вход</router-link>
```
` |` можно не добавлять. В "меню" в App.vue задачи они были для ручного тестирования самой задачи :)

ShGKme reviewed (changes requested):
Вёрстка немного отличается. Это не влияет на суть задачи, но если потом собирать в проект, будет проще, если сразу сделать такую же вёрстку. Я прикрепил к комментариям ссылки на фрагменты кода, чтобы было проще.

aleksandrayakovleva132 committed:
fix layout

aleksandrayakovleva132 committed:
save commit

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

aleksandrayakovleva132 committed:
Update RegisterPage.vue

aleksandrayakovleva132 committed:
Merge branch 'master' of github.com:js-tasks-ru/vue-20210304_aleksandra-yakovleva9

aleksandrayakovleva132 committed:
solution wrappers - button

Задачу с обработкой логина / регистрации отложила, пока не понимаю принцип работы обработки запросов

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

ShGKme commented on 04-spa/04-Layouts/components/AuthLayout.vue:
```diff
@@ -1,8 +1,20 @@
+<template>
+  <div class="page page_onboarding">
+    <div class="container">
+      <h1 class="page__title text-center">Вход</h1>
```
Текст заголовка может быть разным, в зависимости от входного параметра.

ShGKme commented on 04-spa/04-Layouts/components/FormLayout.vue:
```diff
@@ -1,8 +1,21 @@
+<template>
+  <div class="page page_meetup-forms">
+    <div class="container">
+      <h2 class="page__title">Редактирование митапа</h2>
```
Аналогично, текст заголовка может быть разным, в зависимости от входного параметра.

ShGKme commented on 06-wrappers/01-Buttons/components/BaseButton.vue:
```diff
@@ -1,10 +1,29 @@
 <template>
-  <button type="button" class="button"></button>
+  <component :is="tag"
+             type="button" class="button"
```
В целом передавать `type="button"`, чтобы не требовалось отключать `submit` по умолчанию - удобно.

Но в такой реализации этот атрибут также будет передаваться на другие элементы, например, на ссылку `<a>`, а также будет иметь больший приоритет, чем `v-bind="$attrs"` и не позволит установить `type="submit"`

ShGKme commented on 06-wrappers/01-Buttons/components/BaseButton.vue:
```diff
@@ -1,10 +1,29 @@
+
+  created() {
+    this.$attrs;
+  },
```
Что делает этот код?

ShGKme commented on 04-spa/04-Layouts/components/TheHeader.vue:
```diff
@@ -1,4 +1,14 @@
+        /></router-link>
+      </h1>
+    </div>
+  </header>
```
А теперь меню совсем потерялось :(
Можно взять из материалов вебинара.
https://github.com/js-tasks-ru/vue-course-materials/blob/master/04-spa/03-vue-router/src/App.vue#L4-L24

aleksandrayakovleva132 committed:
Register & Login Page

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

aleksandrayakovleva132 committed:
test register

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

aleksandrayakovleva132 committed:
AuthPages

jsru commented:
Добавляю преподавателя (@ShGKme) для код-ревью.

ShGKme commented on 06-wrappers/01-Buttons/components/BaseButton.vue:
```diff
@@ -1,10 +1,29 @@
+      type: String,
+      default: 'button',
+    },
+  },
```
В общем случае тег может быть не только строкой, но и объектом и даже функцией, также как в is у `<component>`. Можно передать имя компонента, а можно передать непосредственно сам компонент.

ShGKme reviewed (changes requested):
Требуется поправить фиксированные заголовки в Layouts на значение заголовка из пропса, вернуть меню в TheHeader и поправить type в Buttons.

aleksandrayakovleva132 committed:
fix code

jsru commented:
Проверьте, пожалуйста, ваше решение, не все тесты прошли (PR не будет принят до тех пор, пока **все** добавленные задачи не будут решены).

aleksandrayakovleva132 committed:
fix code

jsru commented:
Добавляю преподавателя (@ShGKme) для код-ревью.

ShGKme commented on 04-spa/04-Layouts/components/BaseLayout.vue:
```diff
@@ -1,10 +1,10 @@
       <slot />
     </main>
-    <the-footer />
+   <TheFooter />
```
Компоненты в шаблоне можно использовать и в kebab-case, и в CamelCase. Главное выбрать какой-то один силь

ShGKme reviewed (approved):
Все задачи приняты 🎉
